src/base/resource_client.ts

```diff
@@ -117,12 +99,15 @@ class ResourceClient extends ApiClient {
         } while ((shouldRepeat()));
```

Comment:
i know its unrelated, but :]
```suggestion
        } while (shouldRepeat());
```